lib/router-config/src/query_planner.rs

```diff
@@ -7,4 +7,6 @@ pub struct QueryPlannerConfig {
     /// When set to `true` and an incoming request has a `hive-expose-query-plan: true` header, the query plan will be exposed in the response, as part of `extensions`.
     #[serde(default)]
     pub allow_expose: bool,
+    #[serde(default)]
+    pub no_plan_cache: bool,
```

Comment:
I think `no_plan_cache: true` feels like you flip the condition too many times? 

why not just `cache: false` (and then the default is `true`)?

Reply:
Let's not do `cache: boolean`, but be explicit.
What if we add `redis` cache of plans? What if we have an in-memory cache and redis at the same time and we want to disable one of them?

Imo, we should either has an option to disable caching entirely with a config option like `{ cache: { enabled: boolean } }` or `{ cache: { in_memory: { enabled: boolean } } }`.

Reply:
This way we can have options for the cache that applies to in-memory, redis, disk-based whatever and also space for configs specific to each option.

Reply:
So are we going to have seperate cache storage options for each feature instead of having one single cache storage options for all features?
I think one cache storage can be shared by multiple features. So caching can be a boolean flag for query planning.

```yml
cache:
  redis: ...
query_plan:
   cache: false
respobse_cache:
  ...
```

vs
```yml
query_plan:
  cache:
    redis: ...
respobse_cache:
  cache:
    redis: ...
```